### ISSUE TITLE
Unable to explicitly expand/collapse list via UX

### ISSUE THREAD

**Issue by BobDickinson** [indrimuska/jquery-editable-select#19]
I have a case where the user may enter any value into an edit control, but I'd like for them to be able to choose from some common values (sometimes they will enter a custom value, but usually they will choose one of the common ones).

This control seemed like a good solution.  When using this with filter: false, the user can see all of the items in the list (no matter what they may have already typed), and choose one from the list, or enter their own value into the edit control.  So far, so good.

But I have few issues with the way the control works right now (for my use case).  I'd like for the list not to show itself unless the user activates it explicitly, and I'd like the list to show itself whenever the user activates it, regardless of the value currently in the edit control and/or whether it was chosen from the list.  And I'd like for the user to be able to collapse the list without selecting anything ("oh, none of these values are what I want, I'm going to type something").

Right now, when the focus goes to the control, the list is shown (regardless of the value in the edit control) - which I don't like for my use case.  If the user chooses a value from the list, the list goes away, and then the list will not come back when you try to activate it again by clicking on the down-triangle indicator (you have to move the focus away from the control and back, then the list pops back down).  And the only way to make the list go away is to select something from it.

So if there was a mode where the list always and only popped down when the user clicked the indicator, and if the list collapsed when the use clicked the indicator again (not requiring a selection to make the list go away), that would be my ideal functionality.   I realize that means hit-testing on the down-triangle element instead of just activating/deactivating the list based on control focus, and maybe that's out of scope for what this control is intended to do.

**Comment by BobDickinson** [comment]
I fixed this locally.  Let me know if you're interested in a PR.  

I add this function:

    EditableSelectUtility.prototype.isOverDisclosureArrow = function (event) {
        var rightPadding = parseInt(this.es.$input.css('padding-right'));
        var rightBorder = parseInt(this.es.$input.css('border-right-width'));
        return (event.offsetX > (this.es.$input.outerWidth() - (rightPadding + rightBorder)));
    };

And then added these handlers in initializeInput():

            .on('mousemove', function (e) {
                if (that.isOverDisclosureArrow(e)) {
                    that.es.$input.css('cursor', 'default');
                } else {
                    that.es.$input.css('cursor', 'auto');
                }
            })
            .on('mousedown', function (e) {
                if (that.isOverDisclosureArrow(e)) {
                    if (that.es.$input.hasClass('open')) {
                        that.es.hide();
                    } else {
                        that.es.show();
                    }
                }
            })

That treats the disclosure arrow more like a button (when you hover over it you get an appropriate cursor, and when you click on it, the state toggles, period).  This doesn't deal with the list being shown on focus (I'm thinking of adding an option to only show the list on focus if the value is empty).

**Comment by indrimuska** [comment]
Hi @BobDickinson, I completely agree with you. I haven't had this kind of use case, but I'm going to cover them all in the next release.

Since the user may want to choose WHEN the dropdown should open (in your use case you want to open the list by clicking or hovering on the disclosure arrow, but others may want to open the list programmatically), I will add a new `trigger` option that allows to manage this choice: `click`, `focus`, `manual`.

You already have methods to open or close the dropdown manually, so for your users you may have something like this:

```html
<!-- HTML -->
<span id="disclosureArrow"></span>
<select id="mySelect">...</select>
```
```css
/* CSS */
#mySelect { background: none }
#disclosureArrow { float: right; }
```
```javascript
// JavaScript
var mySelect = $('#mySelect').editableSelect({ trigger: 'manual' });

// open/close the dropdown using the disclosure arrow events
$('#disclosureArrow').on('click', function () {
   if (mySelect.is('.open')) mySelect.editableSelect('hide');
   else mySelect.editableSelect('show');
});
```

**Comment by indrimuska** [comment]
FYI, trigger option is now available from [v2.2.0](https://github.com/indrimuska/jquery-editable-select/releases/tag/2.2.0).
